src/cnf/mod.rs

```diff
@@ -25,14 +25,37 @@ pub const SERVER_AGENT: &str = concat!("SurrealDB ", env!("CARGO_PKG_VERSION"));
 #[cfg(feature = "has-storage")]
 pub const APP_ENDPOINT: &str = "https://surrealdb.com/app";
 
-/// How many concurrent tasks can be handled in a WebSocket
-#[cfg(feature = "has-storage")]
-pub const MAX_CONCURRENT_CALLS: usize = 24;
-
 /// Specifies the frequency with which ping messages should be sent to the client
 #[cfg(feature = "has-storage")]
 pub const WEBSOCKET_PING_FREQUENCY: Duration = Duration::from_secs(5);
 
+/// Set the maximum WebSocket frame size to 16mb
+#[cfg(feature = "has-storage")]
+pub static WEBSOCKET_MAX_FRAME_SIZE: Lazy<usize> = Lazy::new(|| {
+	let default = 64 << 20;
+	std::env::var("WEBSOCKET_MAX_FRAME_SIZE")
+		.map(|v| v.parse::<usize>().unwrap_or(default))
+		.unwrap_or(default)
+});
+
+/// Set the maximum WebSocket frame size to 256mb
+#[cfg(feature = "has-storage")]
+pub static WEBSOCKET_MAX_MESSAGE_SIZE: Lazy<usize> = Lazy::new(|| {
+	let default = 256 << 20;
+	std::env::var("WEBSOCKET_MAX_MESSAGE_SIZE")
+		.map(|v| v.parse::<usize>().unwrap_or(default))
+		.unwrap_or(default)
+});
+
+/// How many concurrent tasks can be handled in a WebSocket
+#[cfg(feature = "has-storage")]
+pub static WEBSOCKET_MAX_CONCURRENT_REQUESTS: Lazy<usize> = Lazy::new(|| {
+	let default = 24;
```

Comment:
Is there a good reason we use 24 as the default anyway? I think it's way too low.

Reply:
These are concurrently executing requests. What do you think a fair limit should be?

Reply:
I'm not sure what the ideal number should be but in my experience, on Linux, when spawning actual threads the limiting factor to me (besides memory) seemed to be the configured maximum number of file descriptors per process. Since async tasks do not spawn actual threads, I think the main issue is memory so I think our bounded channels can handle much higher numbers. Because of this experience, I typically set bounded channels to at least 1024.

My experiment back then wasn't inspired by trying to find the optimal performance but rather how many tasks the program could handle concurrently before blowing up, so my assumptions may be wrong. A case in point is [this concurrency example](https://github.com/surrealdb/surrealdb/blob/main/lib/examples/concurrency/main.rs). I set capacity in that example to 100k (which creates bounded channels of that number) because I noticed that the server was able to handle that number without crashing on my average Acer laptop back then. When I set the number much higher (I don't remember the exact number) the server crushed but the SDK itself was still fine. Now I'm able to bump that number to 1m and it still runs fine. I don't know if it's because I'm using a bigger machine now or the server has improved since then (maybe it's both 😄).

Reply:
Definitely @rushmorem! The server can definitely handle more concurrent requests, but from a server perspective should we really allow users to make thousands or millions of concurrent database requests? We can increase this number though, I guess it's about finding the right number where we draw the line!